lib/DxilContainer/DxilContainerAssembler.cpp

```diff
@@ -1592,39 +1592,45 @@ void hlsl::SerializeDxilContainerForModule(DxilModule *pModule,
       // bitcode, which will include the source references, line numbers, etc. Otherwise,
       // do it exclusively on the target shader bitcode.
 
-      pHashStream = (int)(Flags & SerializeDxilFlags::DebugNameDependOnSource)
-                        ? CComPtr<AbstractMemoryStream>(pModuleBitcode)
-                        : CComPtr<AbstractMemoryStream>(pProgramStream);
-
-      // Use user specified debug name if a) it's given and b) it's not a path
-      bool UseDebugName = DebugName.size() && !DebugName.endswith(llvm::StringRef("\\"));
-
       // Calculate the length of the name
-      const uint32_t NameLen = UseDebugName ? 
-        DebugName.size() :
-        DebugInfoNameHashLen +  DebugInfoNameSuffix;
+      const uint32_t NameLen = !DebugName.empty()
```

Comment:
The code was already like that, but it'd be much nicer to just concat the final name string here, use its `.size()`, and have a single lambda that is not conditional on the hash.

Reply:
True.